src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -632,8 +632,16 @@ int32_t SystemNative_FChMod(intptr_t fd, int32_t mode)
 
 int32_t SystemNative_FSync(intptr_t fd)
 {
+    int fileDescriptor = ToFileDescriptor(fd);
+
     int32_t result;
-    while ((result = fsync(ToFileDescriptor(fd))) < 0 && errno == EINTR);
+    while ((result = 
+#if HAVE_F_FULLFSYNC
```

Comment:
From what I understand in the title, we only want to call `fcntrl(fd, F_FULLFSYNC)` on OSX. But with how the code is written right now, if we execute this in any other Unix platform and F_FULLFSYNC is available, we are going to use that since it is being checked first, but I think we should keep using `fsync(fd)` for all other Unix platforms.

I suggest we add an extra check in the `#if` so that we only use `HAVE_F_FULLFSYNC` if we are in the OSX platform.

Reply:
> I suggest we add an extra check in the #if so that we only use HAVE_F_FULLFSYNC if we are in the OSX platform.

Do you know how that check would look like?

Reply:
Since we wouldn't be calling fsync for OSX case, we can also consider adding a new function and then split the implementation on managed code.

Reply:
To check if the code is being executed in OSX, you could use `defined(TARGET_OSX)`. There are many usage examples within `runtime/src/libraries/Native/Unix`.

Reply:
I tighten the condition to use fcntl only for OSX and if F_FULLFSYNC exists, thanks.